packages/admin/admin-date-time/src/TimeRangePicker.tsx

```diff
@@ -0,0 +1,160 @@
+import { ComponentsOverrides, FormControl, Theme, Typography } from "@mui/material";
+import { createStyles, WithStyles, withStyles } from "@mui/styles";
+import clsx from "clsx";
+import * as React from "react";
+import { FormatDateOptions, FormattedMessage } from "react-intl";
+
+import { TimePicker, TimePickerProps } from "./TimePicker";
+
+export type TimeRange = {
+    start: string;
+    end: string;
+};
+
+export type TimeRangePickerIndividualPickerProps = Omit<TimePickerProps, "onChange" | "onBlur" | "value" | "className" | "inputRef">;
+
+export interface TimeRangePickerComponentsProps {
+    startPicker?: TimeRangePickerIndividualPickerProps;
+    endPicker?: TimeRangePickerIndividualPickerProps;
+}
+
+export interface TimeRangePickerProps {
+    onChange?: (timeRange?: TimeRange) => void;
+    value?: TimeRange;
+    formatOptions?: FormatDateOptions;
+    minuteStep?: number;
+    min?: string;
+    max?: string;
+    clearable?: boolean;
+    separatorText?: React.ReactNode;
+    componentsProps?: TimeRangePickerComponentsProps;
+}
+
+type IndividualTimeValue = string | undefined;
+
+function TimeRangePicker({
+    classes,
+    onChange,
+    value,
+    separatorText = <FormattedMessage id="cometAdmin.dateTime.fromToSeparatorText" defaultMessage="to" />,
+    componentsProps = {},
+    ...propsForBothTimePickers
+}: TimeRangePickerProps & WithStyles<typeof styles>) {
+    const [startTime, setStartTime] = React.useState<IndividualTimeValue>(value?.start);
+    const [endTime, setEndTime] = React.useState<IndividualTimeValue>(value?.end);
+
+    const [startPickerIsOpen, setStartPickerIsOpen] = React.useState<boolean>(false);
+    const [endPickerIsOpen, setEndPickerIsOpen] = React.useState<boolean>(false);
+
+    const startPickerRef = React.useRef<HTMLElement>(null);
+    const endPickerRef = React.useRef<HTMLElement>(null);
+
+    React.useEffect(() => {
```

Comment:
I believe you don't need those two useEffects. The respective `onChange` and `setState` calls can be implemented in the change handlers